Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/DriverPool.hs

```diff
@@ -425,6 +428,153 @@ calculateDriverPoolWithActualDist poolCalculationStage driverPoolCfg mbVariant p
+calculateDriverPoolCurrentlyOnRide poolStage driverPoolCfg mbVariant pickup merchantId mRadiusStep reduceRadiousValue = do
+  let radius = getRadius mRadiusStep
+  let coord = getCoordinates pickup
+  now <- getCurrentTime
+  approxDriverPool <-
+    measuringDurationToLog INFO "calculateDriverPoolCurrentlyOnRide" $
+      Esq.runInReplica $
+        QP.getNearestDriversCurrentlyOnRide
+          mbVariant
+          coord
+          radius
+          merchantId
+          driverPoolCfg.driverPositionInfoExpiry
+          reduceRadiousValue
+  maxParallelSearchRequests <- asks (.maxParallelSearchRequests)
+  driversWithLessThanNParallelRequests <- case poolStage of
+    DriverSelection -> filterM (fmap (< maxParallelSearchRequests) . getParallelSearchRequestCount now) approxDriverPool
+    Estimate -> pure approxDriverPool --estimate stage we dont need to consider actual parallel request counts
+  pure $ makeDriverPoolResult <$> driversWithLessThanNParallelRequests
+  where
+    getParallelSearchRequestCount now dObj = getValidSearchRequestCount merchantId (cast dObj.driverId) now
+    getRadius mRadiusStep_ = do
+      let maxRadius = fromIntegral driverPoolCfg.maxRadiusOfSearch
+      case mRadiusStep_ of
+        Just radiusStep -> do
+          let minRadius = fromIntegral driverPoolCfg.minRadiusOfSearch
+          let radiusStepSize = fromIntegral driverPoolCfg.radiusStepSize
+          min (minRadius + radiusStepSize * radiusStep) maxRadius
+        Nothing -> maxRadius
+    makeDriverPoolResult :: QP.NearestDriversResultCurrentlyOnRide -> DriverPoolResultCurrentlyOnRide
+    makeDriverPoolResult QP.NearestDriversResultCurrentlyOnRide {..} =
+      DriverPoolResultCurrentlyOnRide
+        { distanceToPickup = distanceToDriver,
+          ..
+        }
+
+calculateDriverCurrentlyOnRideWithActualDist ::
+  ( EncFlow m r,
+    CacheFlow m r,
+    EsqDBFlow m r,
+    Esq.EsqDBReplicaFlow m r,
+    CoreMetrics m,
+    HasDriverPoolConfig r,
+    HasCoordinates a
+  ) =>
+  PoolCalculationStage ->
+  DriverPoolConfig ->
+  Maybe Variant ->
+  a ->
+  Id DM.Merchant ->
+  Maybe PoolRadiusStep ->
+  Int ->
+  m [DriverPoolWithActualDistResult]
+calculateDriverCurrentlyOnRideWithActualDist poolCalculationStage driverPoolCfg mbVariant pickup merchantId mRadiusStep reduceRadiousValue = do
+  driverPool <- calculateDriverPoolCurrentlyOnRide poolCalculationStage driverPoolCfg mbVariant pickup merchantId mRadiusStep reduceRadiousValue
+  case driverPool of
+    [] -> return []
+    (a : pprox) -> do
+      let y = driverResultFromDestinationLocation <$> (a :| pprox)
+      driverPoolWithActualDistFromDestinationLocation <- computeActualDistance merchantId pickup y
+      driverToDestinationDistanceThreshold <- asks (.driverPoolCfg.driverToDestinationDistanceThreshold)
+      driverPoolWithACtualDistFromCurrentLocation <- sequence $ calculateActualDistanceCurrently driverToDestinationDistanceThreshold <$> (a :| pprox)
```

Comment:
typo `driverPoolWithACtualDistFromCurrentLocation` C is also capital in ACtual.

Reply:
instead of `sequence $ calculateActualDistanceCurrently driverToDestinationDistanceThreshold <$> (a :| pprox)`
you can use `traverse calculateActualDistanceCurrently driverToDestinationDistanceThreshold (a :| pprox)`